### PR TITLE
Get rid of tanukiwrapper dependency for taskomatic / rhn-search

### PR DESCRIPTION
**Pull request by mseidl**
## What does this PR change?
This get's rid of tanukiwrapper and just uses regular shell scripts in systemd.  

This gets rid of the tanuki from spec files, build files, config files.  

There is some changed behavior with taskomatic, we're using systemd to pass an environment file now.  I added a new config file: /etc/rhn/taskomatic.conf  which will be where the customers config options can be added.  

No difference.

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- [doc-susemanager](https://github.com/SUSE/doc-susemanager) PR or issue was created (GitHub automatic link expected below)

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Fixes #
 #1551 (suma)
- [ ] **DONE**

## Changelogs

Copy the following sentence as a new comment if you don't need changelog entries:

> gitarro no changelog needed !!!

If the test `changelog_test`already run, then add another new comment with the following text:

> gitarro rerun changelog_test !!!

### CODE REVIEW

**Comment by mseidl** [comment]
@renner  @cavalheiro @ncounter 

What should we do about the configs?  We're no longer parsing them in a meaningful way.  Most of them started with wrapper.... which isn't good anymore. I've moved most of the jvm stuff into the start scripts.

File: java/code/src/com/redhat/rhn/taskomatic/core/BaseDaemon.java [hunk 1]
```diff
@@ -116,15 +98,30 @@ public void logMessage(int logLevel, String msg, Throwable err) {
...
+                break;
+            case LOG_INFO:
+                LOG.info(buf.toString());
+                break;
+            case LOG_FATAL:
+                LOG.fatal(buf.toString());
+                break;
+            default:
+                break;
+        }
```
[the remark this answers is not in the document]
Comment (mseidl) [reply java/code/src/com/redhat/rhn/taskomatic/core/BaseDaemon.java]: this only called 2x by taskomaticaemon i thought we should just get rid of this myself and call log4j directly.

File: java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java [hunk 2]
```diff
@@ -86,7 +85,7 @@ public void run() {
                         kernel.startup();
                     }
                     catch (Throwable e) {
-                        logMessage(BaseDaemon.LOG_FATAL, e.getMessage(), e);
+                        LOG.fatal(3);
```
Comment (mbologna) [inline java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java]: is it worth to log the message?

**Comment by mseidl** [comment]
> Do we actually need BaseDaemon.java? It looks almost empty now - why not remove it and modify TaskomaticDaemon.java accordingly?

removed.

**Comment by mseidl** [comment]
!gitarro rerun all

File: java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java [hunk 3]
```diff
@@ -15,9 +15,13 @@
 package com.redhat.rhn.taskomatic.core;
 
 import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.PosixParser;
```
Comment (ncounter) [inline java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java]: Maybe you don't have the rule enabled in your IDE but `imports` should be alphabetically ordered.

File: java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java [hunk 4]
```diff
@@ -157,4 +225,26 @@ private void createOption(Options accum, String longopt, boolean arg,
             this.masterOptionsMap.put(longopt, option);
         }
     }
+
+    /**
+     * Registers the daemon implementation with the Tanuki wrapper
```
Comment (ncounter) [inline java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java]: Is this correct to keep mentioning `Tanuki wrapper`?

Comment (mseidl) [reply java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java]: i noticed that and started taking care of it.

File: java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java (hunk 3 above)
Comment (mseidl) [reply java/code/src/com/redhat/rhn/taskomatic/core/TaskomaticDaemon.java]: ok. i'll take care of it.

**Comment by mseidl** [comment]
gitarro rerun !!!

**Comment by mseidl** [comment]
gitarro no javascript_lint !!!

**Pull request merged**